crates/core/src/auth.rs

```diff
@@ -0,0 +1,258 @@
+use crate::error::AuthError;
+use eyre::Result;
+use reqwest::Client;
+use serde::Deserialize;
+use tokio::time::{sleep, Duration};
+
+const BASE_URL: &str = "https://credible-layer-dapp.pages.dev";
+const POLL_INTERVAL: Duration = Duration::from_secs(2);
+const MAX_RETRIES: u32 = 150; // 5 minutes worth of 2-second intervals
+
+const PHYLAX_ASCII: &str = r#"
+@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+@@@@@@@@@@@@@@@@@@BD>"            "<8@@@@@@@@@@@@@@@@@@@
+@@@@@@@@@@@@@@BP"     _.               '4@@@@@@@@@@@@@@@
+@@@@@@@@@@@@D    _e@@B  ,_,   __  t@g_    `G@@@@@@@@@@@@
+@@@@@@@@@@P   _g@@@@P  /@@@  [@@@  \@@@@_    %@@@@@@@@@@
+@@@@@@@@B   _B@@@@@W  {@@@@  [@@@@  T@@@@@a   `@@@@@@@@@
+@@@@@@@P   g@@@@@@@  ;@@@@@  [@@@@A  @@@@@@@_   f@@@@@@@
+@@@@@@P  ,@@@@@@@@F  @@@@@@  g@@@@@  !@@@@@@@L   V@@@@@@
+@@@@@B   @@@@@@@@@  ;@@@@@@  B@@@@@|  @@@@@@@@L   @@@@@@
+@@@@@'  [@@@@@@@@@  g@@BBD>  <4B@@@@  @@@@@@@@@   '@@@@@
+@@@@@   @@@@@@@@@@  BW  __    __ `8@  B@@@@@@@@j   @@@@@
+@@@@@                 ;@@@@  B@@@;                 @@@@@
+@@@@@   qgg@@@@@@g  __ "B@@  @BB" __  g@@@@@@gq;   @@@@@
+@@@@@   @@@@@@@@@@  @@@q___  ___g@@B  @@@@@@@@@   .@@@@@
+@@@@@@   @@@@@@@@@  [@@@@@@  @@@@@@|  @@@@@@@@P   g@@@@@
+@@@@@@\  '@@@@@@@@,  @@@@@@  @@@@@@  |@@@@@@@W   /@@@@@@
+@@@@@@@L  `@@@@@@@@  0@@@@g  @@@@@F  @@@@@@@P   /@@@@@@@
+@@@@@@@@p   \@@@@@@,  @@@@8  @@@@W  A@@@@@B    j@@@@@@@@
+@@@@@@@@@@_   "@@@@@,  @@@]  @@@D  /@@@@D    _@@@@@@@@@@
+@@@@@@@@@@@@_    <B@@_  <=   "8"  /@BP"    _@@@@@@@@@@@@
+@@@@@@@@@@@@@@@_     ""          "      _g@@@@@@@@@@@@@@
+@@@@@@@@@@@@@@@@@@g__              __g@@@@@@@@@@@@@@@@@@
+@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+"#;
+
+#[derive(Deserialize)]
+struct AuthResponse {
+    code: String,
+    #[serde(rename = "sessionId")]
+    session_id: String,
+    #[serde(rename = "deviceSecret")]
+    device_secret: String,
+    #[serde(rename = "expiresAt")]
+    expires_at: String,
+}
+
+#[derive(Deserialize)]
+struct StatusResponse {
+    verified: bool,
+    address: Option<String>,
+    token: Option<String>,
+    refresh_token: Option<String>,
+}
+
+#[derive(clap::Parser)]
+pub struct AuthCommand {
+    #[command(subcommand)]
+    pub command: AuthSubcommands,
+}
+
+#[derive(clap::Subcommand)]
+pub enum AuthSubcommands {
+    /// Login to the CLI
+    Login,
+    /// Logout from the CLI
+    Logout,
+    /// Check current auth status
+    Status,
+}
+
+impl AuthCommand {
+    pub async fn run(&self, config: &mut CliConfig) -> Result<(), AuthError> {
+        match &self.command {
+            AuthSubcommands::Login => self.login(config).await,
+            AuthSubcommands::Logout => self.logout(config),
+            AuthSubcommands::Status => self.status(config),
+        }
+    }
+
+    async fn login(&self, config: &mut CliConfig) -> Result<(), AuthError> {
```

Comment:
The function does way too many things. It should be broken into at least 3 functions:
1. initialize auth session
2. display instructions
3. poll for authentication status

e.g 
```rust
   pub async fn login(&self, config: &mut CliConfig) -> Result<(), AuthError> {
        let client = Client::new();
        
        // Initialize authentication session
        let auth_response = self.initialize_auth_session(&client).await?;
        
        // Display authentication instructions
        self.display_auth_instructions(&auth_response);
        
        // Poll for authentication status
        self.poll_auth_status(&client, &auth_response, config).await
    }
```